src/pegmatite/pattern/repeat.cr

```diff
@@ -8,13 +8,18 @@ module Pegmatite
   # succeeded (not consuming any bytes from the final failing occurrence).
   # Like Pattern::Optional, this pattern will never fail if @min is zero.
   class Pattern::Repeat < Pattern
-    def initialize(@child : Pattern, @min = 0)
+    def initialize(@child : Pattern, @min = 0, max = nil)
+      @max = max || Int32::MAX
     end
 
     def inspect(io)
       @child.inspect(io)
       io << ".repeat("
       @min.inspect(io)
+      if max = @max
```

Comment:
```suggestion
      if max = @max
```

When would `@max` be non-truthy? In the `inititialize` function your code is assigning it `Int32::MAX` as a default if it's non-truthy, so I wouldn't ever expect it to be non-truthy here.

Reply:
Oops you're right. That's a hangover from when I had max as a nilable value